tests/components/mqtt/test_light_template.py

```diff
@@ -320,9 +326,284 @@
+
+
+async def test_sending_mqtt_commands_non_optimistic_brightness_template(
+    hass, mqtt_mock
+):
+    """Test the sending of command in optimistic mode."""
+    with assert_setup_component(1, light.DOMAIN):
+        assert await async_setup_component(
+            hass,
+            light.DOMAIN,
+            {
+                light.DOMAIN: {
+                    "platform": "mqtt",
+                    "schema": "template",
+                    "name": "test",
+                    "effect_list": ["rainbow", "colorloop"],
+                    "state_topic": "test_light_rgb",
+                    "command_topic": "test_light_rgb/set",
+                    "command_on_template": "on,"
+                    "{{ brightness|d }},"
+                    "{{ color_temp|d }},"
+                    "{{ white_value|d }},"
+                    "{{ red|d }}-"
+                    "{{ green|d }}-"
+                    "{{ blue|d }}",
+                    "command_off_template": "off",
+                    "state_template": '{{ value.split(",")[0] }}',
+                    "brightness_template": '{{ value.split(",")[1] }}',
+                    "color_temp_template": '{{ value.split(",")[2] }}',
+                    "white_value_template": '{{ value.split(",")[3] }}',
+                    "red_template": '{{ value.split(",")[4].' 'split("-")[0] }}',
+                    "green_template": '{{ value.split(",")[4].' 'split("-")[1] }}',
+                    "blue_template": '{{ value.split(",")[4].' 'split("-")[2] }}',
+                    "effect_template": '{{ value.split(",")[5] }}',
+                }
+            },
+        )
+
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert not state.attributes.get("brightness")
+    assert not state.attributes.get("hs_color")
+    assert not state.attributes.get("effect")
+    assert not state.attributes.get("color_temp")
+    assert not state.attributes.get("white_value")
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+    await common.async_turn_off(hass, "light.test")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "off", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+
+    await common.async_turn_on(hass, "light.test")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,,--", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+
+    # Set color_temp
+    await common.async_turn_on(hass, "light.test", color_temp=70)
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,70,,--", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert not state.attributes.get("color_temp")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5199018)